site/ja/tutorials/text/transformer.ipynb

```diff
@@ -0,0 +1,2077 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "BSV3PPKsYecw"
+   },
+   "outputs": [],
+   "source": [
+    "class MultiHeadAttention(tf.keras.layers.Layer):\n",
+    "  def __init__(self, d_model, num_heads):\n",
+    "    super(MultiHeadAttention, self).__init__()\n",
+    "    self.num_heads = num_heads\n",
+    "    self.d_model = d_model\n",
+    "    \n",
+    "    assert d_model % self.num_heads == 0\n",
+    "    \n",
+    "    self.depth = d_model // self.num_heads\n",
+    "    \n",
+    "    self.wq = tf.keras.layers.Dense(d_model)\n",
+    "    self.wk = tf.keras.layers.Dense(d_model)\n",
+    "    self.wv = tf.keras.layers.Dense(d_model)\n",
+    "    \n",
+    "    self.dense = tf.keras.layers.Dense(d_model)\n",
+    "        \n",
+    "  def split_heads(self, x, batch_size):\n",
+    "    \"\"\"最後の次元を(num_heads, depth)に分割。\n",
+    "    結果をshapeが(batch_size, num_heads, seq_len, depth)となるようにリシェイプする。\n",
+    "    \"\"\"\n",
+    "    x = tf.reshape(x, (batch_size, -1, self.num_heads, self.depth))\n",
+    "    return tf.transpose(x, perm=[0, 2, 1, 3])\n",
+    "    \n",
+    "  def call(self, v, k, q, mask):\n",
+    "    batch_size = tf.shape(q)[0]\n",
+    "    \n",
+    "    q = self.wq(q)  # (batch_size, seq_len, d_model)\n",
+    "    k = self.wk(k)  # (batch_size, seq_len, d_model)\n",
+    "    v = self.wv(v)  # (batch_size, seq_len, d_model)\n",
+    "    \n",
+    "    q = self.split_heads(q, batch_size)  # (batch_size, num_heads, seq_len_q, depth)\n",
+    "    k = self.split_heads(k, batch_size)  # (batch_size, num_heads, seq_len_k, depth)\n",
+    "    v = self.split_heads(v, batch_size)  # (batch_size, num_heads, seq_len_v, depth)\n",
+    "    \n",
+    "    # scaled_attention.shape == (batch_size, num_heads, seq_len_q, depth)\n",
+    "    # attention_weights.shape == (batch_size, num_heads, seq_len_q, seq_len_k)\n",
+    "    scaled_attention, attention_weights = scaled_dot_product_attention(\n",
+    "        q, k, v, mask)\n",
+    "    \n",
+    "    scaled_attention = tf.transpose(scaled_attention, perm=[0, 2, 1, 3])  # (batch_size, seq_len_q, num_heads, depth)\n",
+    "\n",
+    "    concat_attention = tf.reshape(scaled_attention, \n",
+    "                                  (batch_size, -1, self.d_model))  # (batch_size, seq_len_q, d_model)\n",
+    "\n",
+    "    output = self.dense(concat_attention)  # (batch_size, seq_len_q, d_model)\n",
+    "        \n",
+    "    return output, attention_weights"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "0D8FJue5lDyZ"
+   },
+   "source": [
+    "試しに、`MultiHeadAttention`レイヤーを作ってみましょう。シーケンス`y`の各位置において、`MultiHeadAttention`はシーケンスのその位置以外の全ての位置に対して8つのアテンション・ヘッドを実行し、各位置それぞれで同じ長さの新しいベクトルを返します。"
```

Comment:
[suggestion]
```
シーケンス`y`の各位置において、`MultiHeadAttention`はシーケンスのその位置以外の全ての位置に対して
8つのアテンション・ヘッドを実行し、各位置それぞれで同じ長さの新しいベクトルを返します。
=====>
シーケンス `y` の各位置において、`MultiHeadAttention` はシーケンスのすべての位置に対して8つのヘッドを用いて
アテンションを計算し、各位置それぞれで同じ長さの新しいベクトルを返します。
```

For each location `i`, if `MultiHeadAttention` runs attention heads all other locations excepting the location `i`, `MultiHeadAttention` returns a new vector of shorter length.

Reply:
Oh! So that's it.